java/fury-core/src/main/java/org/apache/fury/serializer/collection/GuavaCollectionSerializers.java

```diff
@@ -399,6 +400,26 @@ public static void registerDefaultSerializers(Fury fury) {
     fury.registerSerializer(cls, new ImmutableSortedSetSerializer<>(fury, cls));
     cls = loadClass(pkg + ".ImmutableSortedMap", ImmutableSortedMap.of(1, 2).getClass());
     fury.registerSerializer(cls, new ImmutableSortedMapSerializer<>(fury, cls));
+
+    // Guava version before 19.0, of() return
+    // EmptyImmutableSet/EmptyImmutableBiMap/EmptyImmutableSortedMap/EmptyImmutableSortedSet
+    // we register if class exist
+    if (checkClassExist(pkg + ".EmptyImmutableSet")) {
```

Comment:
If caller and server have different guava version, this method seems introduce deserialization error

Reply:
Could we make classes before and after guava 19.0 mapped to same class id. In this way, teh different guava version across processes won't introduce deserialization error

Reply:
> Could we make classes before and after guava 19.0 mapped to same class id. In this way, teh different guava version across processes won't introduce deserialization error

I perfer it more, with different guava version, it can take `of()` to deserialize.

Reply:
If classid are different, the deserialization will laod wrong class and deserializer. We met such errors before. it's very hard to locate it

Reply:
> If classid are different, the deserialization will laod wrong class and deserializer. We met such errors before. it's very hard to locate it

I see comments, and I take same classId to deserialize low version guava.